File: t/perl/PushJump.pl

use strict;
use warnings;

use Sm::TestClass;

my $obj = new Sm::TestClass();
$obj->{_fsm}->setDebugStream(\*STDOUT);
$obj->{_fsm}->setDebugFlag(@ARGV > 0);
$obj->Evt_1();
$obj->Evt_2(); # push
$obj->Evt_2(); # pop
$obj->Evt_2(); # push
$obj->Evt3(1); # pop
$obj->Evt_1();
$obj->Evt3(1); # jump
$obj->Evt_1(); # jump
$obj->Evt_1();

File: t/ruby/PushJump.rb

require 'TestClass'

obj = Sm::TestClass::new
obj.getFSM().setDebugStream($stdout)
obj.getFSM().setDebugFlag(ARGV.size > 0)
obj.Evt_1()
obj.Evt_2() # push
obj.Evt_2() # pop
obj.Evt_2() # push
obj.Evt3(1) # pop
obj.Evt_1()
obj.Evt3(1) # jump
obj.Evt_1() # jump
obj.Evt_1()
